tests/integration-e2e/test_dry_run_workflow.py

```diff
@@ -1,16 +1,27 @@
+import json
 import platform
 import pytest
 import os
+import tempfile
 from pathlib import Path
 from unittest.mock import patch
 from neurodamus.utils.memory import DryRunStats
 from neurodamus.core.configuration import GlobalConfig, LogLevel
 
 SIM_DIR = Path(__file__).parent.parent.absolute() / "simulations"
+TMP_FOLDER = tempfile.mkdtemp()
```

Comment:
when more tests are added to this file, does this mean that they will use the same test folder? I think that might be confusing.

Reply:
Each invocation of pytest uses its own path for `tmppath` / `tmpdir` fixtures: https://docs.pytest.org/en/stable/how-to/tmp_path.html#temporary-directory-location-and-retention which can be controlled with `--basetemp`.

In this file, however, since it doesn't use pytest's `tmpdir`, it will pick a different location, afaik pytest doesn't override this, so it's not controllable.

I now remember that @WeinaJi mentioned that this had to do w/ how the tests are structured, and inter-depencencies.  I think it would be worthwhile to restructure the tests so it's not the case - but maybe that needs to be a in a different PR.

Reply:
I improved the comment, and will summarize this PR change in one note.